Assets/HoloToolkit/Speech/Scripts/TextToSpeechManager.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+using System;
+using UnityEngine;
+
+#if WINDOWS_UWP
+using Windows.Media.SpeechSynthesis;
+using Windows.Storage.Streams;
+using System.Threading.Tasks;
+#endif
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Enables text to speech using the Windows 10 <see cref="SpeechSynthesizer"/> class.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="SpeechSynthesizer"/> generates speech as a <see cref="SpeechSynthesisStream"/>. 
+    /// This class converts that stream into a Unity <see cref="AudioClip"/> and plays the clip using 
+    /// the <see cref="AudioSource"/> you supply in the inspector. This allows you to position the voice 
+    /// as desired in 3D space. One recommended approach is to place the AudioSource on an empty 
+    /// GameObject that is a child of Main Camera and position it approximately 0.6 units above the 
+    /// camera. This orientation will sound similar to Cortanas speech in the OS.
+    /// </remarks>
+    public class TextToSpeechManager : Singleton<TextToSpeechManager>
```

Comment:
I personally still believe that the most common use case is going to be a single "omnipotent" voice ala Cortana. And I believe the fastest route to getting that is

1. Place the audio source
2. Link the audio source to speech component
3. Call Speak in code

I do feel that inheriting from Singleton<T> makes that scenario as easy as possible without breaking the ability to support multiple instances. This is possible because even though it inherits from Singleton<T>, the way Singleton<T> is written in the library it still allows multiple instances (only one is the 'default' instance).

Having said that, I also realize that this is confusing. In "regular" programming languages you can't have more than one singleton. It's because of the way the Singleton<T> class in the framework is currently designed that allows it to work. To me, this is another one of those odd things about the way Unity development differs from traditional C# development and I'm okay with the flexibility. However if we feel this difference is too confusing the I propose we just remove Singleton<T> as the base class and just have it inherit from MonoBehaviour. The downside is that the user will have to manually find a reference to the component before having to call it, but it's not a huge deal. We could also create a new class in the framework called DefaultInstance<T> that is effectively implemented the same way that Singleton<T> is but instead of a .Instance property it would have a .Default property. This would allow for the "shortcut" path to the default instance without being confusing by calling it a "Singleton".

Please let me know what design you guys would be comfortable with. I would like to update my code and pull request and see if we can get it accepted.

Thanks everyone for taking the time to review and provide feedback. I really do appreciate it.